### DISCUSSION TITLE
[BUG] [DOCKER] Frigate crash after upgrade

### DISCUSSION THREAD

**Discussion by salaun29** [blakeblackshear/frigate#14195] [General Support] [answered]
Hello,

After upgrading from "ghcr.io/blakeblackshear/frigate:350abda" to any other last version (Frigate 0.15.0-350abda), I have the following issue :

> 2024-10-07 14:32:46.859523387  [INFO] Starting Frigate...
> 
> s6-rc: info: service nginx successfully started
> s6-rc: info: service certsync: starting
> s6-rc: info: service certsync successfully started
> s6-rc: info: service legacy-services: starting
> 2024-10-07 14:32:47.010923217  [INFO] Starting certsync...
> s6-rc: info: service legacy-services successfully started
> Traceback (most recent call last):
>   File "/usr/local/go2rtc/create_config.py", line 17, in <module>
>     from frigate.ffmpeg_presets import parse_preset_hardware_acceleration_encode
>   File "/opt/frigate/frigate/ffmpeg_presets.py", line 70, in <module>
>     FFMPEG_HWACCEL_VAAPI: f"-hwaccel_flags allow_profile_mismatch -hwaccel vaapi -hwaccel_device {_gpu_selector.get_selected_gpu()} -hwaccel_output_format vaapi",
>   File "/opt/frigate/frigate/ffmpeg_presets.py", line 35, in get_selected_gpu
>     self._selected_gpu = f"/dev/dri/{devices[0]}"
> IndexError: list index out of range
> 2024-10-07 14:32:47.173004093  127.0.0.1 - - [07/Oct/2024:14:32:47 +0200] "" 400 0 "-" "-" "-"
> 2024-10-07 14:32:47.207577611  [INFO] The go2rtc service exited with code 1 (by signal 0)
> 2024-10-07 14:32:48.213384969  [INFO] Preparing new go2rtc config...
> 2024-10-07 14:32:48.219743763  Traceback (most recent call last):
> 2024-10-07 14:32:48.219755274    File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
> 2024-10-07 14:32:48.219758841      return _run_code(code, main_globals, None,
> 2024-10-07 14:32:48.219761356    File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
> 2024-10-07 14:32:48.219763670      exec(code, run_globals)
> 2024-10-07 14:32:48.219765213    File "/opt/frigate/frigate/__main__.py", line 9, in <module>
> 2024-10-07 14:32:48.219802201      from frigate.app import FrigateApp
> 2024-10-07 14:32:48.219804084    File "/opt/frigate/frigate/app.py", line 18, in <module>
> 2024-10-07 14:32:48.219805767      from frigate.api.auth import hash_password
> 2024-10-07 14:32:48.219807110    File "/opt/frigate/frigate/api/auth.py", line 27, in <module>
> 2024-10-07 14:32:48.219808382      from frigate.config import AuthConfig, ProxyConfig
> 2024-10-07 14:32:48.219809624    File "/opt/frigate/frigate/config/__init__.py", line 4, in <module>
> 2024-10-07 14:32:48.219851803      from .camera import *  # noqa: F403
> 2024-10-07 14:32:48.219853205    File "/opt/frigate/frigate/config/camera/__init__.py", line 3, in <module>
> 2024-10-07 14:32:48.219873272      from .camera import *  # noqa: F403
> 2024-10-07 14:32:48.219874664    File "/opt/frigate/frigate/config/camera/camera.py", line 7, in <module>
> 2024-10-07 14:32:48.219875947      from frigate.ffmpeg_presets import (
> 2024-10-07 14:32:48.219877169    File "/opt/frigate/frigate/ffmpeg_presets.py", line 70, in <module>
> 2024-10-07 14:32:48.219879203      FFMPEG_HWACCEL_VAAPI: f"-hwaccel_flags allow_profile_mismatch -hwaccel vaapi -hwaccel_device {_gpu_selector.get_selected_gpu()} -hwaccel_output_format vaapi",
> 2024-10-07 14:32:48.219880706    File "/opt/frigate/frigate/ffmpeg_presets.py", line 35, in get_selected_gpu
> 2024-10-07 14:32:48.219881768      self._selected_gpu = f"/dev/dri/{devices[0]}"
> 2024-10-07 14:32:48.219882790  IndexError: list index out of range
> 2024-10-07 14:32:48.250068726  2024/10/07 14:32:48 [error] 183#183: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 172.18.0.1, server: , request: "GET /api/stats HTTP/1.1", subrequest: "/auth", upstream: "http://127.0.0.1:5001/auth", host: "100.78.9.23:5000"
> 2024-10-07 14:32:48.250074797  2024/10/07 14:32:48 [error] 183#183: *2 auth request unexpected status: 502 while sending to client, client: 172.18.0.1, server: , request: "GET /api/stats HTTP/1.1", host: "100.78.9.23:5000"
> Traceback (most recent call last):
>   File "/usr/local/go2rtc/create_config.py", line 17, in <module>
>     from frigate.ffmpeg_presets import parse_preset_hardware_acceleration_encode
>   File "/opt/frigate/frigate/ffmpeg_presets.py", line 70, in <module>
>     FFMPEG_HWACCEL_VAAPI: f"-hwaccel_flags allow_profile_mismatch -hwaccel vaapi -hwaccel_device {_gpu_selector.get_selected_gpu()} -hwaccel_output_format vaapi",
>   File "/opt/frigate/frigate/ffmpeg_presets.py", line 35, in get_selected_gpu
>     self._selected_gpu = f"/dev/dri/{devices[0]}"
> IndexError: list index out of range
> 2024-10-07 14:32:48.534069588  [INFO] Service Frigate exited with code 1 (by signal 0)
> s6-rc: info: service legacy-services: stopping

Any idea on how to solve this ?

N.B:Reverting back to 350abda solve the issue :)

Thx,

**Comment by NickM-27** [comment]
this was already fixed

**Reply by salaun29** [reply]
Hi,

Thx for the fast answer but I have the same issue with the latest version dcaed0e.

Regards,

**Reply by NickM-27** [reply]
that's not the latest version https://github.com/blakeblackshear/frigate/commit/74047453ef08f6d4b81da6173fb164016990843e

**Reply by salaun29** [reply]
Thanks for the fix done but It is still not working. but no more crash now frigate start :)

> 2024-10-07 17:19:18.324355731  [2024-10-07 17:19:18] frigate.util.services          ERROR   : Unable to poll vainfo: b"error: XDG_RUNTIME_DIR is invalid or not set in the environment.\nerror: can't connect to X server!\nlibva info: VA-API version 1.22.0\nlibva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/bochs-drm_drv_video.so\nlibva info: va_openDriver() returns -1\nvaInitialize failed with error code -1 (unknown libva error),exit\n"
> 2024-10-07 17:19:18.381817126  [2024-10-07 17:19:18] frigate.util.services          ERROR   : Unable to poll intel GPU stats: No device filter specified and no discrete/integrated i915 devices found
> 2024-10-07 17:19:18.381822627

**Reply by NickM-27** [reply, marked as answer]
the bug only happened when no hardware was present. Things are working correctly, this is a docker config issue

**Reply by salaun29** [reply]
Thanks a lot, you rock
